src/ddmd/delegatize.d

```diff
@@ -170,12 +170,23 @@ extern (C++) bool lambdaCheckForNestedRef(Expression e, Scope* sc)
     return v.result;
 }
 
-bool checkNestedRef(Dsymbol s, Dsymbol p)
+/*****************************************
+ * See if context `s` is nested within context `p`, meaning
+ * it `p` is reachable at runtime by walking the static links.
+ * If any of the intervening contexts are function literals,
+ * make sure they are delegates.
+ * Params:
+ *      s = inner context
+ *      p = outer context
+ * Returns:
+ *      true means it is accessible by walking the context pointers at runtime
+ */
+bool ensureStaticLinkTo(Dsymbol s, Dsymbol p)
```

Comment:
How about `ensureContextIsReachable`? The term "static links" requires some very specific [terminology knowledge](https://en.wikipedia.org/wiki/Call_stack#Functions_of_the_call_stack) and is tricky to search. I'd rather see "static link" in the documentation and "reachable context" in the name, than vice versa.